torch/csrc/jit/passes/specialize_autogradzero.cpp

```diff
@@ -147,7 +147,7 @@ struct AutogradZeroSpecializer {
 
       state_[inp] = *pttp->undefined() ? State::Zero : State::Nonzero;
       auto check = graph_->insert(prim::AutogradAnyNonZero, {inp});
-      if (!*pttp->undefined()) {
+      if (*pttp->undefined()) {
```

Comment:
Btw, do we need checks for not-undefined tensors at all? Wouldn't checking only undefined ones suffice?

Reply:
We do optimizations based on both the defined & undefined state of Tensor, so it's good to specialize both